src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+using SafeCurlHandle = Interop.libcurl.SafeCurlHandle;
+using CURLoption = Interop.libcurl.CURLoption;
+using CURLcode = Interop.libcurl.CURLcode;
+using CURLINFO = Interop.libcurl.CURLINFO;
+
+namespace System.Net.Http
+{
+    internal class CurlHandler : HttpMessageHandler
+    {
+        #region Constants
+
+        private const string UriSchemeHttps = "https";
+
+        #endregion
+
+        #region Fields
+
+        private volatile bool _anyOperationStarted;
+        private volatile bool _disposed;
+        private bool _automaticRedirection = true;
+
+        #endregion
+
+        internal CurlHandler()
+        {
+        }
+
+        #region Properties
+        internal bool AutomaticRedirection
+        {
+            get
+            {
+                return _automaticRedirection;
+            }
+
+            set
+            {
+                CheckDisposedOrStarted();
+                _automaticRedirection = value;
+            }
+        }
+
+        #endregion
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && !_disposed)
+            {
+                _disposed = true;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        protected internal override Task<HttpResponseMessage> SendAsync(
+            HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException("request", SR.net_http_handler_norequest);
+            }
+            if (request.RequestUri.Scheme == UriSchemeHttps)
```

Comment:
According to http://curl.haxx.se/libcurl/c/CURLOPT_URL.html there are a lot of protocols supported by libcurl. You should enforce either UriSchemeHttp or UriSchemeHttps and reject anything else for HttpClient.